api/v1alpha1/turbine_types.go

```diff
@@ -0,0 +1,77 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TurbineSpec defines the desired state of Turbine
+type TurbineSpec struct {
+	// Sidecar configuration
+	Sidecar *SidecarConfiguration `json:"sidecar,omitempty"`
```

Comment:
To fill the default values of `SideCar` field we should give default values to it. At least one default field will allow other default values to be filled by the API server. It would be best to give an empty object but kubebuilder does not support empty object default value.
```suggestion
	// +kubebuilder:default:={Name:  "turbine-sidecar"}
	Sidecar *SidecarConfiguration `json:"sidecar,omitempty"`
```